rpcs3/Emu/System.cpp

```diff
@@ -1794,12 +1795,20 @@ void Emulator::GracefulShutdown(bool allow_autoexit, bool async_op)
 
 	auto perform_kill = [allow_autoexit, this, info = ProcureCurrentEmulationCourseInformation()]()
 	{
-		for (u32 i = 0; i < 100; i++)
+		bool read_sysutil_signal = false;
+
+		for (u32 i = 200; i < 250; i++)
 		{
-			std::this_thread::sleep_for(50ms);
+			std::this_thread::sleep_for(20ms);
 			Resume(); // TODO: Prevent pausing by other threads while in this loop
 			process_qt_events(); // Is nullified when performed on non-main thread
 
+			if (!read_sysutil_signal && !are_sysutil_callbacks_pending())
+			{
+				i -= 200; // Grant 4 seconds (if signal is not read force kill after one second)
```

Comment:
it was 5 seconds before though....

Reply:
A second is an enormous amount of time to execute a sysutil callback. Most games read signals each frame and even more frequently.
The 5s span was about executing _sys_process_exit which typically occurs much later.

Reply:
some games take a ridiculous amount of time to exit.
I don't get what you're trying to achieve by decreasing this time by a second

Reply:
executing cellSysutilChecCallback() != executing _sys_process_exit()

Reply:
If one second passed and it hadn't picked up the signal already it probably won't at all.

Reply:
This isn't about the 4 seconds you decrease.
It's about the 1 second you took from the main loop

Reply:
actually you took 4 seconds, which is even worse

Reply:
5 seconds feel a lot when games crash or freeze internally and can't react to the signal. A lot of times especially in initial intros of games they had yet to read sysutil signals repeatedly as well. And some games do not read them in a loop at all.

Reply:
I'd rather wait 5 seconds for my shit game to die than have the good games die prematurely and potentially lose progress.

Reply:
You misunderstand the reaction time of PPU, the reaction time is very short. If it was so lengthy and RSX VBLANK signals would be read every let's assume half a second because of the supposed lengthy PPU reaction time, games would not reach more than 2fps.

Reply:
Then read this section and tell me what happens when the game receives the exit_cmd and needs 4 seconds to do some jank auto_save.

Reply:
The behavior is the same as before if it picks up the sysutil signal (5s span), most of the time until Emu.Kill() is called is spent **after** the sysutil signal has been picked up.

Reply:
You tell the game to die in 1789.
Then you wait one second if the game called checkCallback quickly.
Then you perform the Kill in 1821.

How do the 4 seconds of jank autosave fit in?

Reply:
If it has called checkCallback it grants it 4 more seconds, that's what the line in question is about.

Reply:
this assumes that the game doesn't create more callbacks....

Reply:
Fixed and increased waiting time to a total of 7 seconds.